mnist_test.go

```diff
@@ -0,0 +1,59 @@
+package gotorch_test
+
+import (
+	"fmt"
+	"log"
+
+	torch "github.com/wangkuiyi/gotorch"
+	nn "github.com/wangkuiyi/gotorch/nn"
+	F "github.com/wangkuiyi/gotorch/nn/functional"
+)
+
+type MultiLayerMNISTNet struct {
+	FC1, FC2, FC3 nn.Module
+}
+
+func (n *MultiLayerMNISTNet) Forward(x torch.Tensor) torch.Tensor {
+	x = torch.View(x, []int{-1, 28 * 28})
+	x = n.FC1.Forward(x)
+	x = torch.Relu(x)
+	x = n.FC2.Forward(x)
+	x = torch.Relu(x)
+	x = n.FC3.Forward(x)
+	return x.LogSoftmax(1)
+}
+
+func NewMNISTNet() nn.Module {
+	return &MultiLayerMNISTNet{
+		FC1: nn.Linear(28*28, 512, false),
+		FC2: nn.Linear(512, 512, false),
+		FC3: nn.Linear(512, 10, false),
+	}
+}
+
+func ExampleTrainMNIST() {
+	if e := downloadMNIST(); e != nil {
+		log.Printf("Cannot find or download MNIST dataset: %v", e)
+	}
+	net := NewMNISTNet()
+	transforms := []torch.Transform{torch.NewNormalize(0.1307, 0.3081)}
+	mnist := torch.NewMNIST(dataDir(), transforms)
+	trainLoader := torch.NewDataLoader(mnist, 64)
+	opt := torch.SGD(0.1, 0, 0, 0, false)
+	opt.AddParameters(nn.GetParameters(net))
+	batchIdx := 0
+	for trainLoader.Scan() {
+		batch := trainLoader.Batch()
+		pred := net.Forward(batch.Data)
+		loss := F.NllLoss(pred, batch.Target, torch.Tensor{nil}, -100, "mean")
+		opt.ZeroGrad()
+		loss.Backward()
```

Comment:
Currently, this example doesn't converge. I think you will need a line `opt.Step` here.

Reply:
fixed, and updated the active function to `Tanh`. Now it converged as the following figure.
![image](https://user-images.githubusercontent.com/1426912/89847317-4c5e4a00-dbb6-11ea-8abd-749e61a8f894.png)

Reply:
I will port the [item](https://pytorch.org/docs/stable/tensors.html#torch.Tensor.item) function in Go so that we can assert the loss value in the unit test.